File: Java/HashMap/Maximum Difference Between Even and Odd Frequency I.java
class Solution {
    public int maxDifference(String s) {
        HashMap<Character,Integer> fq=new HashMap<>();
        for(char ch:s.toCharArray()){
            fq.put(ch,fq.getOrDefault(ch,0)+1);
        }
        int xodd=0;
        int neven=s.length();
        for(Integer i:fq.values()){
            if(i%2==1)xodd=Math.max(xodd,i);
            else
            neven=Math.min(neven,i);
        }
        return xodd-neven;
    }
}

File: Python/dictionary/Maximum Difference Between Even and Odd Frequency I.py
class Solution:
    def maxDifference(self, s: str) -> int:
        fq={}
        for ch in s:
            fq[ch]=fq.get(ch,0)+1
        
        xodd=0
        neven=len(s)
        for i in fq:
            if fq[i]%2==1:
                xodd=max(xodd,fq[i])
            else:
                neven=min(neven,fq[i])

        return xodd-neven
